### PR TITLE
Feature/naming and optimizations

### PR DESCRIPTION
**Pull request by hihiben**
- [x] Remove redundant word fee after charge
- [x] Use calc instead of calculate
- [x] Remove underscore prefix in library internals
- [x] Rename IParam to DataType
- [x] Rename DUST to DUST_IN_WEI
- [x] Rearrange function order

### CODE REVIEW

File: src/Router.sol [hunk 1]
```diff
@@ -444,26 +467,19 @@ contract Router is IRouter, Ownable, EIP712 {
...
-    /// @param pauser_ The pauser address
-    function setPauser(address pauser_) public onlyOwner {
-        if (pauser_ == address(0)) revert InvalidNewPauser();
-        pauser = pauser_;
-        emit PauserSet(pauser_);
-    }
+    /// @notice Invalidate nonces for an execution
+    /// @param newNonce The new nonce to set. Invalidates all nonces less than it
+    /// @dev Can't invalidate more than 2**16 nonces per transaction
+    function invalidateExecutionNonces(uint256 newNonce) external {
```
[the remark this answers is not in the document]
Comment (hihiben) [reply src/Router.sol:473]: I assume that `BySig` execution is also delegation lol. Can change to this though.

[the remark this answers is not in the document]
Comment (Jeff-CCH) [reply src/Router.sol:473]: suggest removing `related`. Signature nonce management is clear enough.

File: src/Router.sol [hunk 2]
```diff
@@ -86,10 +86,67 @@ contract Router is IRouter, Ownable, EIP712 {
         transferOwnership(owner_);
     }
 
-    /// @notice Get owner address
-    /// @return The owner address
-    function owner() public view override(IRouter, Ownable) returns (address) {
-        return super.owner();
+    ///////////////////////////////////////////////////////////////////////////////////////
+    /// Router related
```
Comment (Jeff-CCH) [inline src/Router.sol:90]: `Router related -> Router Operations`
Is it easier to understand?

File: src/AgentImplementation.sol [hunk 3]
```diff
@@ -62,11 +62,6 @@ contract AgentImplementation is IAgent, ERC721Holder, ERC1155Holder {
         permit2 = permit2_;
     }
 
-    /// @notice Return current fee charging status when calling
-    function isCharging() external view returns (bool) {
-        return _callbackWithCharge.isCharging();
-    }
-
```
Comment (Jeff-CCH) [inline src/AgentImplementation.sol:69]: Which coding style do we follow for the function order? Do we follow the public order or the group order?

Public order: order functions in a file
Group order: order functions in groups

File: src/Router.sol [hunk 4]
```diff
@@ -98,6 +155,40 @@ contract Router is IRouter, Ownable, EIP712 {
...
     }
 
+    /// @notice Get owner address
+    /// @return The owner address
+    function owner() public view override(IRouter, Ownable) returns (address) {
+        return super.owner();
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////////////
+    /// Agent related
```
Comment (Jeff-CCH) [inline src/Router.sol:165]: `Agent related -> Agent Operations`
Is it easier to understand?

File: src/Router.sol [hunk 5]
```diff
@@ -134,41 +225,9 @@ contract Router is IRouter, Ownable, EIP712 {
...
-    }
-
-    /// @notice Unpause `execute` and `executeWithSignerFee` by pauser
-    function unpause() external onlyPauser {
-        if (currentUser != _PAUSED) revert NotPaused();
-        currentUser = _INIT_CURRENT_USER;
-        emit Unpaused();
-    }
+    ///////////////////////////////////////////////////////////////////////////////////////
+    /// Execution related
```
Comment (Jeff-CCH) [inline src/Router.sol:229]: `Execution related -> Executions`
Is it easier to understand?

File: src/Router.sol [hunk 6]
```diff
@@ -351,45 +394,25 @@ contract Router is IRouter, Ownable, EIP712 {
...
     }
 
-    function _newAgent(address user) internal returns (address) {
-        IAgent agent = IAgent(address(new Agent{salt: bytes32(bytes20(user))}(agentImplementation)));
-        agents[user] = agent;
-        emit AgentCreated(address(agent), user);
-        return address(agent);
-    }
+    ///////////////////////////////////////////////////////////////////////////////////////
+    /// Delegation management related
```
Comment (Jeff-CCH) [inline src/Router.sol:414]: Delegation management related -> Delegation Management

File: src/Router.sol [hunk 7]
```diff
@@ -444,26 +467,19 @@ contract Router is IRouter, Ownable, EIP712 {
...
+    /// @notice Invalidate nonces for an execution
+    /// @param newNonce The new nonce to set. Invalidates all nonces less than it
+    /// @dev Can't invalidate more than 2**16 nonces per transaction
+    function invalidateExecutionNonces(uint256 newNonce) external {
+        uint256 oldNonce = executionNonces[msg.sender];
+        if (newNonce <= oldNonce) revert InvalidNonce();
+        // Limit the amount of nonces that can be invalidated in one transaction.
+        unchecked {
+            uint256 delta = newNonce - oldNonce;
+            if (delta > type(uint16).max) revert ExcessiveInvalidation();
```
Comment (Jeff-CCH) [inline src/Router.sol:479]: just found we may be able to skip `uint256 delta = newNonce - oldNonce` and use `if (newNonce - oldNonce > type(uint16).max)` instead?

Comment (hihiben) [reply src/Router.sol:479]: In fact it's from [permit2](https://github.com/Uniswap/permit2/blob/main/src/AllowanceTransfer.sol#L120C20-L120C25). But yes, we can do that.

File: src/libraries/DataType.sol [hunk 8]
```diff
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: GPL-3.0-or-later
 pragma solidity ^0.8.0;
 
-interface IParam {
+library DataType {
```
[the remark this answers is not in the document]
Comment (hihiben) [reply src/libraries/DataType.sol:4]: Will rearrange the order in this library also.

File: src/Router.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (hihiben) [reply src/Router.sol:473]: In that case do we move both `invalidateDelegationNonces` and `invalidateExecutionNonces` to this section?

File: src/AgentImplementation.sol (hunk 3 above)
Comment (hihiben) [reply src/AgentImplementation.sol:69]: Group order is in higher priority when it helps to classify the function into different groups for better readability. However the grouping might not be trivial so additional discussion is needed. When the contract is not too complicated, applying public order should be the most straightforward way to do the work.

**Review by Jeff-CCH — APPROVED** [review]
LGTM

**Review by zodahu — APPROVED** [review]
LGTM

**Pull request merged**